armi/reactor/converters/uniformMesh.py

```diff
@@ -131,11 +131,12 @@ def convert(self, r=None):
                 f"the core's reference assembly mesh: {r.core.refAssem.getAxialMesh()}"
             )
             self.convReactor = self._sourceReactor
+            self.convReactor.core.updateAxialMesh()
             self._setParamsToUpdate()
             for assem in self.convReactor.core.getAssemblies(self._nonUniformMeshFlags):
                 homogAssem = self.makeAssemWithUniformMesh(
                     assem,
-                    self.convReactor.core.refAssem.getAxialMesh(),
+                    self.convReactor.core.p.axialMesh[1:],
```

Comment:
The purpose of this PR is to map new quantities from a uniform mesh back to a non-uniform mesh.

This is not immediately related to the `pinQuantities` category, but it is required for a downstream plugin to generate the `pinQuantities` parameters. It should probably be a separate one-line PR, but it is a bug that I uncovered while working on this one.

`refAssem.getAxialMesh()` gives the block mesh, while `axialMesh[1:]` is the full axial mesh including block subdivisions (i.e., `b.p.axMesh`). Since we always set `b.p.axMesh = 1` in `makeAssemWithUniformMesh` (https://github.com/terrapower/armi/blob/bc3cf475b2158bc3be626287b18f2e21d5da52a7/armi/reactor/converters/uniformMesh.py#L380), we should use `axialMesh[1:]` as the mesh to homogenize onto.

Reply:
I moved this to #959

Reply:
Ah, you uncovered a bug.

Okay.